videosink/handler.go

```
@@ -0,0 +1,179 @@
+	}
+
+	return buf.Bytes(), nil
+}
+
+func (d *Display) grabSnapshot(cfg imageConfig) ([]byte, error) {
+	var err error
+
+	d.mu.Lock()
+	defer d.mu.Unlock()
+
+	encoded, ok := d.snapshot[cfg]
+	if !ok {
+		encoded, err = d.encodeBufferLocked(cfg.format)
+		if err != nil {
+			return nil, err
+		}
+		d.snapshot[cfg] = encoded
+	}
+
+	return append(bufferPool.Get().([]byte)[:0], encoded...), nil
+}
+
+// ServeHTTP handles HTTP GET requests and sends a stream of images
+// representing the display buffer in response. The display options control the
+// default format and clients can explicitly request PNG or JPEG images using
+// the "format" parameter ("?format=png", "?format=jpeg").
+func (d *Display) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	if err := r.Body.Close(); err != nil {
+		log.Printf("Closing request body failed: %v", err)
+	}
+
+	if r.Method != http.MethodGet {
+		http.Error(w, "", http.StatusMethodNotAllowed)
+		return
+	}
+
+	cfg, err := d.configFromQuery(r.URL.Query())
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	pw := makePartWriter(w)
+
+	w.Header().Set("Content-Type",
+		mime.FormatMediaType("multipart/x-mixed-replace", map[string]string{
+			"boundary": pw.boundary,
+		}))
+
+	c := &client{
+		refresh:   make(chan struct{}, 1),
+		terminate: make(chan struct{}, 1),
+	}
+
+	d.mu.Lock()
+	d.clients[c] = struct{}{}
+	d.mu.Unlock()
+
+	defer func() {
+		d.mu.Lock()
+		delete(d.clients, c)
+		d.mu.Unlock()
+	}()
+
+	partHeaders := make(textproto.MIMEHeader)
+	partHeaders.Set("Content-Type", mime.FormatMediaType(cfg.format.mimeType(), nil))
+	partHeaders.Set("Content-Transfer-Encoding", "binary")
+
+	for {
+		payload, err := d.grabSnapshot(cfg)
+		if err == nil {
+			err = pw.writeFrame(partHeaders, payload)
+			//lint:ignore SA6002 buffer is []byte and thus pointer-like
+			bufferPool.Put(payload)
+		}
+
+		if err != nil {
+			log.Printf("Request failed: %v", err)
```

I would prefer no log statement here.

---

What would you prefer to have instead? `fmt.Fprintf(os.Stderr, …)`, a configurable logger function/interface in the options, or silently ignoring?

---

Add a comment for now and silently ignore. We can add later. I prefer the code to not log by default.

---

Done.